dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -70,66 +72,162 @@ class Identity[A] extends Transformer[A, A] {
-      private var labelTensor: Tensor[T] = Tensor[T]()
+      private val featureTensor: Tensor[T] = Tensor[T]()
+      private val labelTensor: Tensor[T] = Tensor[T]()
       private var featureData: Array[T] = null
       private var labelData: Array[T] = null
+      private var sampleData: Array[Sample[T]] = null
+
       private val batchSize = Utils.getBatchSize(totalBatch)
       private var featureSize: Array[Int] = null
       private var labelSize: Array[Int] = null
       private var oneFeatureLength: Int = 0
       private var oneLabelLength: Int = 0
+      private var padFeatureLength: Int = 0
+      private var padLabelLength: Int = 0
+      private val padFeature: Boolean = if (padFeatureValue.isEmpty) false else true
+      private val padLabel: Boolean = if (padLabelValue.isEmpty) false else true
       override def hasNext: Boolean = prev.hasNext
 
       override def next(): MiniBatch[T] = {
         if (prev.hasNext) {
           var i = 0
+          var (maxfeatureElement, labelIndex) = (0, 0)
+          var (maxlabelElement, featureIndex) = (0, 0)
+          var padLength: Int = 0
+          if (sampleData == null) sampleData = new Array[Sample[T]](batchSize)
           while (i < batchSize && prev.hasNext) {
             val sample = prev.next()
-            if(!sample.feature().isContiguous() || !sample.label().isContiguous()) {
-              throw new IllegalArgumentException(
-                "Only support contiguous tensor, pls use tensor.contiguous() before batching")
+            require(sample.feature().isContiguous() && sample.label().isContiguous(),
+              "SampleToBatchPadding: Only support contiguous tensor, pls use " +
+                "tensor.contiguous() before batching")
+            if (sampleData(i) == null) sampleData(i) = Sample()
+            if ((maxfeatureElement != 0) && (!padFeature)) {
+              require(sample.feature().nElement() == maxfeatureElement,
+                "SampleToBatchPadding: all feature size should be same when not padding feature")
             }
-            if (featureData == null) {
-              oneFeatureLength = sample.feature().nElement()
-              oneLabelLength = sample.label().nElement()
-              /**
-               * For example,
-               * the featureSize of an image is: [3, 224, 224]
-               * the labelSize of an image is: [1]
-               *
-               * the featureSize of a fixed sentence input is: [10, 1000]
-               * the labelSize of a fixed sentence input is: [10]
-               */
-              featureSize = Array(1) ++ sample.feature().size
-              labelSize = Array(1) ++ sample.label().size
-              featureData = new Array[T](batchSize * oneFeatureLength)
-              labelData = new Array[T](batchSize * oneLabelLength)
+            if ((maxlabelElement != 0) && (!padLabel)) {
+              require(sample.label().nElement() == maxlabelElement,
```

Comment:
The check is not correct. Ideally we need to check if all labels in the batch have the same size (instead of same element#).